Do not print out status table when specifying --format json

MasterOdin opened the pull request:
Sort of fixes #962

If using `--format json` for `phinx status`, it will print out just the JSON object and not the full status table as well.

Before:
```
Phinx by CakePHP - https://phinx.org. 0.11.1

using config file ./phinx.yml
using config parser yaml
using migration paths
 - /Users/mpeveler/Work/Github/phinx/db/migrations
using seed paths
warning no environment specified, defaulting to: production
using format json
ordering by creation time

 Status  [Migration ID]  Started              Finished             Migration Name
----------------------------------------------------------------------------------
     up  20190828175258  2019-08-28 17:55:41  2019-08-28 17:55:41  TestMigration

{"pending_count":0,"missing_count":0,"total_count":1,"migrations":{"20190828175259":{"migration_status":"up","migration_id":"20190828175258","migration_name":"TestMigration"}}}
```

After:
```
Phinx by CakePHP - https://phinx.org. 0.11.1

using config file ./phinx.yml
using config parser yaml
using migration paths
 - /Users/mpeveler/Work/Github/phinx/db/migrations
using seed paths
warning no environment specified, defaulting to: production
using format json
ordering by creation time

{"pending_count":0,"missing_count":0,"total_count":1,"migrations":{"20190828175259":{"migration_status":"up","migration_id":"20190828175258","migration_name":"TestMigration"}}}
```

Working on a separate PR/proposal for how to deal with the header text in a nicer fashion making it easier to use this (and other) command in a pipeline without having to use tail or grep.

dereuromark commented on tests/Phinx/Migration/ManagerTest.php:
```diff
@@ -164,6 +164,45 @@ public function testPrintStatusMethod()
+        $this->manager->setEnvironments(['mockenv' => $envStub]);
+        $this->manager->getOutput()->setDecorated(false);
+        $return = $this->manager->printStatus('mockenv', 'json');
+        $this->assertEquals(0, $return);
```
ideally Same() to check also Type.

MasterOdin commented:
So it looks like there's a bug where if the keys of the `$migrations` array are integers it gets printed one way versus if they're a string they get printed another way. The former is what's happening on Travis and the latter is on Windows.

PHP should (normally) automatically cast any string key that looks like an integer into an integer, e.g.:
```
$a = ["1" => 'a'];
var_dump(array_keys($a));
```
should give: 
```
array(1) {
  [0] =>
  int(1)
}
```

but on windows it's giving:
```
array(1) {
  [0] =>
  string(1)
}
```

MasterOdin commented:
Realized from #1602 what's going on Windows and why there's a string. It's because the build there is 32-bit and the date strings are past the value of MAX_PHP_INT for 32bit and so have to be used as a strings whereas Travis uses 64-bit and is happy to store things as integers with its much wider range.

I guess the only real solution here is to just fix whatever issue exists in printStatus so that it behaves the same regardless if the keys of the array are integer or string.

dereuromark commented:
Changes make sense to me. For format we do not want to mess it up, otherwise keep it as is. :+1:

dereuromark commented:
Would be good to have @lorenzo s feedback before merging, though.

lorenzo commented on src/Phinx/Migration/Manager.php:
```diff
@@ -158,7 +160,9 @@ public function printStatus($environment, $format = null)
-                    $this->printMissingVersion($missingVersion, $maxNameLength);
+                    if ($format !== 'json') {
+                        $this->printMissingVersion($missingVersion, $maxNameLength);
+                    }
```
Given how much this code is repeated, can we add the check for the format in the function itself?

MasterOdin replied:
Yeah, these if statements were a clumsy way to go about it. I've implemented a vastly simpler solution.

dereuromark commented:
ping @MasterOdin

MasterOdin commented:
Sorry, I have ended up being busy the last few weeks with moving to a new country and starting a new job.

The main thing currently blocking me is:

> I guess the only real solution here is to just fix whatever issue exists in printStatus so that it behaves the same regardless if the keys of the array are integer or string.

So that behavior is consistent across 32 and 64 bit Phinx for printing out the status array.

dereuromark commented:
@lorenzo Are there any hard blockers? Or can we also do follow up PRs here terms of further improving this?

MasterOdin commented:
@dereuromark @lorenzo looking at this further, and running a couple of local tests, I think actually the 64-bit format is a _regression_ on 32-bit and that this behavior should be fixed. I've created #1612 to document the current behavior on the two platforms as I was able to get it, and a case that clearly (to me) demonstrates a bug in the 64-bit format.

I would argue that fixing #1612 is a blocker to this PR being merged so that I can write a single test for behavior for this change, and that resolving the merge conflicts between the two will not be that significant. Assuming you folks agree, I'm going to focus on trying to fix #1612 first before addressing the current merge conflict and @lorenzo's comments.

MasterOdin commented:
Alright, with #1612 done and not having to fight with the output of appveyor vs travis, I've addressed the review comment here, vastly simplifying my original approach to something way more maintainable.

@dereuromark, @lorenzo this should be ready for review!
